pkg/network/tracer/tracer_test.go

```diff
@@ -1881,3 +1883,169 @@ func skipIfWindows(t *testing.T) {
+        self.send_response(status_code)
+        self.end_headers()
+
+server_address = ('127.0.0.1', 8001)
+httpd = http.server.HTTPServer(server_address, RequestHandler)
+httpd.socket = ssl.wrap_socket(httpd.socket,
+                               server_side=True,
+                               certfile='%s',
+                               keyfile='%s',
+                               ssl_version=ssl.PROTOCOL_TLS)
+httpd.serve_forever()
+`
+
+func TestOpenSSLVersions_python_matrix(t *testing.T) {
+	if !httpSupported(t) {
+		t.Skip("HTTPS feature not available on pre 4.14.0 kernels")
+	}
+
+	disableTLSVerification.Do(func() {
+		nethttp.DefaultTransport.(*nethttp.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	})
+	curDir, _ := testutil.CurDir()
+	crtPath := filepath.Join(curDir, "testdata/cert.pem.0")
+	keyPath := filepath.Join(curDir, "testdata/server.key")
+	pythonSSLServer := fmt.Sprintf(pythonSSLServerFormat, crtPath, keyPath)
+	scriptFile, err := writeTempFile("python_openssl_script", pythonSSLServer)
+	require.NoError(t, err)
+	defer scriptFile.Close()
+
+	cmd := exec.Command("python3", scriptFile.Name())
+	go func() {
+		err := cmd.Start()
+		if err != nil {
+			fmt.Println(err)
+		}
+	}()
+	defer func() {
+		if cmd.Process != nil {
+			cmd.Process.Kill()
+		}
+	}()
+
+	cfg := config.New()
+	cfg.EnableHTTPSMonitoring = true
+	cfg.EnableHTTPMonitoring = true
+	cfg.BPFDebug = true
+	tr, err := NewTracer(cfg)
+	require.NoError(t, err)
+	err = tr.RegisterClient("1")
+	require.NoError(t, err)
+	defer tr.Stop()
+
+	// Waiting for the server to be ready
+	portCtx, cancelPortCtx := context.WithDeadline(context.Background(), time.Now().Add(time.Second*5))
+	rawConnect(portCtx, t, "127.0.0.1", "8001")
+	cancelPortCtx()
+
+	requestFn := simpleGetRequestsGenerator(t, "127.0.0.1:8001")
+	var requests []*nethttp.Request
+	for i := 0; i < 100; i++ {
+		requests = append(requests, requestFn())
+	}
+
+	assertAllRequestsExists(t, tr, requests)
+}
+
+var (
+	statusCodes = []int{nethttp.StatusOK, nethttp.StatusMultipleChoices, nethttp.StatusBadRequest, nethttp.StatusInternalServerError}
+)
+
+func simpleGetRequestsGenerator(t *testing.T, targetAddr string) func() *nethttp.Request {
+	var (
+		random = rand.New(rand.NewSource(time.Now().Unix()))
+		idx    = 0
+		client = new(nethttp.Client)
+	)
+
+	return func() *nethttp.Request {
+		idx++
+		status := statusCodes[random.Intn(len(statusCodes))]
+		req, err := nethttp.NewRequest(nethttp.MethodGet, fmt.Sprintf("https://%s/%d/request-%d", targetAddr, status, idx), nil)
+		require.NoError(t, err)
+		resp, err := client.Do(req)
+		require.NoError(t, err)
+		require.Equal(t, status, resp.StatusCode)
+		resp.Body.Close()
+		return req
+	}
+}
+
+func assertAllRequestsExists(t *testing.T, tracer *Tracer, requests []*nethttp.Request) {
+	requestsExist := make([]bool, len(requests))
+	for i := 0; i < 10; i++ {
+		time.Sleep(10 * time.Millisecond)
```

Comment:
Why the code sleeps here?

Reply:
to give time for the kernel to flush the buffers to the usermode

Reply:
Please add a comment about it for future us/others

Reply:
added